src/Clock.tsx

```diff
@@ -0,0 +1,47 @@
+/* eslint-disable no-console */
+import { Component } from 'react';
+
+type Props = {
+  clockName: string,
+};
+
+type State = {
+  date: Date;
+};
+
+export class Clock extends Component<Props, State> {
+  timerId?: number;
+
+  state = {
+    date: new Date(),
+  };
+
+  componentDidMount() {
+    this.timerId = window.setInterval(() => {
+      this.setState({ date: new Date() });
+    }, 1000);
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    console.log(`Renamed from ${prevProps.clockName} to ${this.props.clockName}`);
```

Comment:
You don't need to print this msg every time the component updates, only if old name !== new name

Reply:
<img width="1440" alt="image" src="https://user-images.githubusercontent.com/52370890/178911354-a3d70370-adfe-49b1-8a10-718565fae5e0.png">